### PR TITLE
Add a postfix to binaries to make mixed packaging easier

### PR DESCRIPTION
**Pull request by sarthakpati**
**CMAKE_DEBUG_POSTFIX** added for Windows builds. This makes packaging+distribution of Debug and Release builds easier.

cc @malfet @seemethere

### CODE REVIEW

**Review by peterjc123 — COMMENTED** [review]
Yes, that's great! But I guess you'll need to change `FindTorch.cmake` too. Otherwise, it will break the linkage to the debug version of LibTorch DLLs.

**Comment by peterjc123** [comment]
The related file: https://github.com/pytorch/pytorch/blob/master/cmake/TorchConfig.cmake.in.
An example from CMake: https://gitlab.kitware.com/cmake/cmake/merge_requests/2398/diffs#c02662467bc732b07e07137a73750ae1ae0b6ca3_45_53

**Comment by sarthakpati** [comment]
@peterjc123 Done! https://github.com/sarthakpati/pytorch/commit/a81224456003640ec94094310585e4463dca5876

File: cmake/TorchConfig.cmake.in [hunk 1]
```diff
@@ -41,19 +41,37 @@ if (@BUILD_SHARED_LIBS@)
...
+  endif()
 else()
   find_library(TORCH_LIBRARY NO_CMAKE_FIND_ROOT_PATH torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
 endif()
 add_library(torch UNKNOWN IMPORTED)
-set(TORCH_LIBRARIES torch ${Caffe2_MAIN_LIBS})
+if (NOT WIN32)
+  set(TORCH_LIBRARIES torch ${Caffe2_MAIN_LIBS})
+else()
+  set(TORCH_LIBRARIES ${TORCH_LIBRARY_RELEASE} ${TORCH_LIBRARY_DEBUG} ${Caffe2_MAIN_LIBS})
```
Comment (peterjc123) [inline cmake/TorchConfig.cmake.in]: I don't think linking against both the debug and release libraries will work here. You should use `select_library_configurations` as an alternative.

File: cmake/TorchConfig.cmake.in [hunk 2]
```diff
@@ -41,19 +41,37 @@ if (@BUILD_SHARED_LIBS@)
...
+    find_library(C10_LIBRARY_RELEASE c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  endif()
 else()
   find_library(C10_LIBRARY c10 NO_CMAKE_FIND_ROOT_PATH PATHS "${TORCH_INSTALL_PREFIX}/lib")
 endif()
-list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
+if (NOT WIN32)
+  list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
+else()
+  list(APPEND TORCH_LIBRARIES ${C10_LIBRARY_DEBUG} ${C10_LIBRARY_RELEASE})
```
Comment (peterjc123) [inline cmake/TorchConfig.cmake.in]: and here

File: cmake/TorchConfig.cmake.in [hunk 3]
```diff
@@ -79,8 +97,14 @@ if (@USE_CUDA@)
...
   endif()
-  find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
-  list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY})
+  if (NOT WIN32)
+    find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY})
+  else()
+    find_library(C10_CUDA_LIBRARY_DEBUG c10_cudad PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(C10_CUDA_LIBRARY_RELEASE c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY_DEBUG} ${C10_CUDA_LIBRARY_RELEASE})
```
Comment (peterjc123) [inline cmake/TorchConfig.cmake.in]: here, too.

**Review by peterjc123 — CHANGES_REQUESTED** [review]
I don't think they are correct. BTW, you'll have to do that for Caffe2 libraries too.

**Pull request closed without merging**